src/main/java/reactor/core/publisher/MonoProcessor.java

```diff
@@ -295,7 +295,7 @@ public final void onNext(O value) {
 		if(value != null) {
 			finalState = STATE_SUCCESS_VALUE;
 			this.value = value;
-			if (s != null) {
+			if (s != null && !(source instanceof Mono)) {
```

Comment:
source can be a `Flux`, as in [here](https://github.com/reactor/reactor-core/blob/38b2cf6953bae554ee81bc2c05df1d87d2373be4/src/main/java/reactor/core/publisher/Flux.java#L4906-L4906)